newIDE/app/src/UI/Notification/NotificationList.js

```diff
@@ -0,0 +1,120 @@
+import ScrollView from '../ScrollView';
+
+type Props = {|
+  notifications: Notification[],
+  onMarkAllAsRead: () => Promise<void>,
+  canMarkAllAsRead: boolean,
+  onMarkNotificationAsSeen: Notification => Promise<void>,
+  onCloseNotificationList: () => void,
+|};
+
+const NotificationList = ({
+  notifications,
+  onMarkAllAsRead,
+  canMarkAllAsRead,
+  onMarkNotificationAsSeen,
+  onCloseNotificationList,
+}: Props) => {
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [showAll, setShowAll] = React.useState<boolean>(false);
+
+  const markAllAsRead = React.useCallback(
+    async () => {
+      setIsLoading(true);
+      try {
+        await onMarkAllAsRead();
+      } catch (error) {
+        console.error(
+          'An error occurred while marking all notifications as seen:',
+          error
+        );
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [onMarkAllAsRead]
+  );
+
+  const markNotificationAsSeen = React.useCallback(
+    async (notification: Notification) => {
+      try {
+        await onMarkNotificationAsSeen(notification);
+      } catch (error) {
+        console.error(
+          'An error occurred while marking notification as seen:',
+          error
+        );
+      }
+    },
+    [onMarkNotificationAsSeen]
+  );
+
+  const notificationsToDisplay = showAll
+    ? notifications
+    : notifications.slice(0, 5);
+
+  return (
+    <I18n>
+      {({ i18n }) => (
+        <Column noMargin>
+          <Line justifyContent="space-between" alignItems="center" noMargin>
+            <Text size="block-title">
+              <Trans>Notifications</Trans>
+            </Text>
+            <LeftLoader isLoading={isLoading}>
+              <FlatButton
+                primary
+                label={<Trans>Mark all as read</Trans>}
+                disabled={!canMarkAllAsRead}
+                onClick={markAllAsRead}
+              />
+            </LeftLoader>
+          </Line>
+          <ScrollView autoHideScrollbar>
+            <Column>
+              <List>
+                {notificationsToDisplay.length > 0 ? (
+                  notificationsToDisplay.map(notification => (
+                    <NotificationListItem
+                      key={notification.id}
+                      notification={notification}
+                      onCloseNotificationList={onCloseNotificationList}
+                      onMarkNotificationAsSeen={() => {
+                        markNotificationAsSeen(notification);
+                      }}
+                    />
+                  ))
+                ) : (
+                  <ListItem
+                    primaryText={<Trans>You have 0 notifications.</Trans>}
+                  />
+                )}
+              </List>
+              <TextButton
+                primary
+                label={<Trans>Load more...</Trans>}
+                onClick={() => setShowAll(true)}
+                disabled={showAll}
```

Comment:
should the button disappear once everything is displayed?

Reply:
Good idea, and it will match the mockups 😅